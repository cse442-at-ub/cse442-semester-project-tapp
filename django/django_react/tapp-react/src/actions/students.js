import axios from 'axios';

import { GET_EVENTS, ADD_EVENT, GET_ERRORS } from './types';

//GET STUDENTS
export const getEvents = (param) => dispatch => {
  axios.get('http://localhost:3000/api/class?classNum='+param)
  .then(res => {
    dispatch({
      type: GET_EVENTS,
      payload: res.data
    });
  })
};


//ADD USER
export const addEvent = events => dispatch => {
  axios	
  .post('/api/class', events)
  .then(res => {
  dispatch({
    type: ADD_EVENT,
    payload: res.data
  });
  })
  .catch(err => {
  dispatch({
    type: GET_ERRORS, 
    payload: err
    });
  });
};
