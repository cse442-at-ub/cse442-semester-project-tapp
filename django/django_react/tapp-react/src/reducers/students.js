import {GET_STUDENTS, ADD_USER} from "../actions/types.js"

const initialState = {
  students: []
}

export default function(state=initialState, action) {
  switch(action.type) {
    case GET_STUDENTS:
      return {
      ...state,
      students: action.payload
      };
    case ADD_USER:
      return {
      ...state,
      students: [...state.students, action.payload]};
    default: return state
  }
}
