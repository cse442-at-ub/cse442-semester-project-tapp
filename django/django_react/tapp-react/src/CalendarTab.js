import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Calendar, momentLocalizer } from 'react-big-calendar';
import moment from 'moment';
import 'react-big-calendar/lib/css/react-big-calendar.css';
import DateTime from 'react-datetime';
import './DateTime.css';
import Row from 'react-bootstrap/Row'
import Col from 'react-bootstrap/Col'
import Container from 'react-bootstrap/Container'
import Button from 'react-bootstrap/Button'
import Alert from 'react-bootstrap/Alert'
import {getEvents, addEvent} from './actions/students';

export class CalendarTab extends Component{
  constructor(props) {
    super(props);
    this.state = {
	    startTime: "",
	    endTime: "",
	    date: "",
            invalidAlert: false,
    }
    this.onChangeStart = this.onChangeStart.bind(this);
    this.onChangeEnd = this.onChangeEnd.bind(this);
    this.onChangeDate = this.onChangeDate.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  componentDidMount() {
    if(this.props.course != null)
    {
      this.props.getEvents(this.props.course);
      ;
    }
  }

  onChangeStart = e => this.setState({ startTime: e});
  onChangeEnd = e => this.setState({ endTime: e });
  onChangeDate = e => this.setState({ date: e});
  handleSubmit = e => {
	  if(!moment(this.state.date, 'YYYY-MM-DD', true).isValid)
          {
		  this.setState({invalidAlert:true});
	  }
	  else{
	  this.props.addEvent({startTime:moment(this.state.date.format('YYYY-MM-DD')+" "+moment(this.state.startTime).format("HH:mm"), 'YYYY-MM-DD HH:mm'), endTime:moment(this.state.date.format('YYYY-MM-DD')+" "+moment(this.state.endTime).format("HH:mm"), 'YYYY-MM-DD HH:mm'), classNum:this.props.course, allDay:false});
	  }
  }

  render() { 
    return (
	    <>
  <div id="tab">
    <Container style={{backgroundColor:"#F5F9E9",borderRadius:"5px"}} fluid>
    <Row className="align-middle justify-content-md-center" style={{paddingTop:"15px",borderRadius:"5px"}} > 
    <Calendar
      localizer={momentLocalizer(moment)}
      defaultDate={moment().toDate()}
      events={[]}
      startAccessor="start"
      endAccessor="end"
      events= {this.props.events.map(myevent => (
	      {
	        title:"Office hours",
		start:moment(myevent.startTime).toDate(),
		end: moment(myevent.endTime).toDate()
	      }))}
      style={{ height: 700 }}
    / >
    </Row>
    <Row className="align-middle justify-content-md-center mt-3">
    <h3> Instructor commands </h3>
    </Row>
    <Row className="align-middle justify-content-md-center mt-3">
  <Alert variant="danger" show = {this.state.invalidAlert} onClose = {() => this.setState({invalidAlert:false})} dismissible>
  Invalid Event entered.
  </Alert>
    </Row>
    <Row style={{paddingBottom:"20px"}}>
    <Col className="flex-column">
    <h5> Event date:   </h5>
    <DateTime inputProps={{placeholder: 'MM/DD/YYYY'}} onChange={this.onChangeDate} style={{height:"50px"}} timeFormat={false}/> 
    </Col>
    <Col className="flex-column">
    <h5> Time start:   </h5>
    <DateTime onChange={this.onChangeStart} dateFormat={false} inputProps={{placeholder: 'HH:MM AM/PM'}}/> 
    </Col>
    <Col className="flex-column">
    <h5> Time end:   </h5>
    <DateTime onChange={this.onChangeEnd} dateFormat={false} inputProps={{placeholder: 'HH:MM AM/PM'}}/> 
    </Col>
    </Row>
    <Row style={{paddingBottom:"100px"}}>
    <Col>
    <Button onClick={this.handleSubmit}> Submit </Button>
    </Col>
    </Row>
    </Container>
  </div>
  </>
    )
  }
}

Calendar.propTypes = {
        addEvent: PropTypes.func.isRequired,
        events: PropTypes.array.isRequired,
        getEvents: PropTypes.func.isRequired
}

const mapStateToProps = state => ({
  events: state.students.events,
});

export default connect(mapStateToProps, {getEvents, addEvent} )(CalendarTab);
