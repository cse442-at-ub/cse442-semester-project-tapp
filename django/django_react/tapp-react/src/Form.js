import React, { Component } from 'react';
import Button from 'react-bootstrap/Button';
import Form from 'react-bootstrap/Form';
import Modal from 'react-bootstrap/Modal';
import Alert from 'react-bootstrap/Alert';
import PropTypes from 'prop-types';
import { addUser } from './actions/students';
import { register } from './actions/auth';
import { connect } from 'react-redux';
import { Link, Redirect } from "react-router-dom";

import 'bootstrap/dist/css/bootstrap.min.css'
import "./styles.css";

export class SignupModal extends Component {
    constructor(props) {
    super(props)
    this.state = {
	    first: false,
            signedUp: false,
            signedValid: false,
            isLoading: false,
            onHide: props.onHide,
            show: props.show,
	    emailValid: false,
	    passValid: false,
	    nameValid: false,
	    courseValid: false,
	    emailMessage: "Please enter an email!",
	    nameMessage: "",
	    courseMessage: "",
	    passMessage: ""
    };
    this.toggleSignedUp = this.toggleSignedUp.bind(this);
    this.toggleSignedUpBool = this.toggleSignedUpBool.bind(this);
    this.toggleSignedValid = this.toggleSignedValid.bind(this);
    this.toggleIsLoading = this.toggleIsLoading.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.checkValidity = this.checkValidity.bind(this);
  }

  componentDidUpdate(prevProps) {
    const {error, alert } = this.props;
    if(error !== prevProps.error) {
     if(this.props.error.msg.email) {this.setState({emailValid: false,emailMessage: this.props.error.msg.email.join()})}
     else {this.setState({emailValid:true,emailMessage:""})}
     if(this.props.error.msg.password) {this.setState({passValid: false, passMessage: this.props.error.msg.pass.join()})}
     else {this.setState({passValid:true,passMessage:""})}
     if(this.props.error.msg.name) {this.setState({nameValid: false, nameMessage: this.props.error.msg.name.join()})}
     else {this.setState({nameValid: true,nameMessage: ""})}
     if(this.props.error.msg.name) {this.setState({courseValid: false, courseMessage: this.props.error.msg.course.join()})}
     else {this.setState({courseValid:true,courseMessage:""})}
    }
  }

  toggleSignedUp = () => {
    this.setState({
      signedUp: !this.state.signedUp
    });
  };

  toggleSignedUpBool = bool => {
    this.setState({
      signedUp: bool
    });
  };

  toggleSignedValid = () => {
    this.setState({
      signedValid: !this.state.signedValid
    });
  };

  toggleIsLoading = () => {
    this.setState({
      isLoading: !this.state.isLoading
    });
  };

  checkValidity = () => {
    //return this.state.emailValid && this.state.nameValid && this.state.passValid;};
    return this.state.emailValid && this.state.nameValid && this.state.courseValid};

  handleSubmit = event => {
	  const form = event.currentTarget;
	  //if (this.checkValidity() === false) { event.preventDefault(); event.stopPropagation();}
	  //else{
          event.preventDefault();
	  if (!this.state.first) {this.setState({first:!this.state.first})};
	  this.setState({emailMessage:"",emailValid:true,nameValid:true,courseValid:true});
	  //this.toggleIsLoading();
	  const name = form.elements.name.value;
	  const email = form.elements.email.value;
	  const password = form.elements.password.value;
	  const course = form.elements.course.value;
	  const instructor = true;
	  const user = { name,email,password,instructor,course };
	  console.log(user);
	  this.props.register(user);
	  if (this.checkValidity() === true)
          {
	    this.toggleSignedUp();
	    this.toggleSignedValid(true);
	  }
  };

  render() {
    if (this.props.isAuth) {
      return <Redirect to="/dashboard" />;
    }
    const signedUp = this.state.signedUp;
    const isLoading = this.state.isLoading;
    return (
      <Modal onHide= {this.props.onHide} show= {this.props.show} size="lg" centered>
        <Modal.Header closeButton>
          <Modal.Title> Signup </Modal.Title>
        </Modal.Header>
        <Modal.Body>
          <Alert variant="success" show={signedUp} dismissible>
            <Alert.Heading> An email verification has been sent to your email </Alert.Heading>
          </Alert>
          <Form onSubmit={this.handleSubmit} validated={this.checkValidity()}>
            <Form.Group controlId="formBasicEmail">
              <Form.Label> E-mail </Form.Label>
              <Form.Control type="email" name="email" placeholder="example@domain.com" isInvalid={!this.state.emailValid&&this.state.first} required />
              <Form.Control.Feedback type="invalid"> {this.state.emailMessage} </Form.Control.Feedback> 
            </Form.Group>
            <Form.Group controlId="formBasicPassword">
              <Form.Label> Password </Form.Label>
              <Form.Control name="password" type="password" placeholder="Password" isInvalid={!this.state.passValid&&this.state.first} required/>
              <Form.Control.Feedback type="invalid"> Please enter a password! </Form.Control.Feedback> 
            </Form.Group>
            <Form.Group controlId="formBasicText">
              <Form.Label> Name </Form.Label>
              <Form.Control name="name" type="text" placeholder="Name" isValid={this.state.nameValid} isInvalid={!this.state.nameValid&&this.state.first} required/>
              <Form.Control.Feedback type="invalid"> Please enter a Name! </Form.Control.Feedback> 
            </Form.Group>
            <Form.Group controlId="formBasicText">
              <Form.Label> Course </Form.Label>
              <Form.Control type="text" name="course" placeholder="Course" isValid={this.state.courseValid} isInvalid={!this.state.courseValid&&this.state.first} required />
              <Form.Control.Feedback type="invalid"> Please enter a Course Name! </Form.Control.Feedback> 
            </Form.Group>
            <Button variant="primary" type="submit" disabled={isLoading}> {isLoading ? 'Loading...':'Submit'} </Button>
          </Form>
        </Modal.Body>
      <Modal.Footer>
        <Button onClick={this.props.onHide}> Close </Button>
      </Modal.Footer>
    </Modal>
    );
  }
}

SignupModal.propTypes = {
  register: PropTypes.func.isRequired,
  isAuth: PropTypes.bool,
  error: PropTypes.object.isRequired
};

const mapStateToProps = state => ({
  error: state.errors
});

export default connect(mapStateToProps,{register, addUser }) (SignupModal);
