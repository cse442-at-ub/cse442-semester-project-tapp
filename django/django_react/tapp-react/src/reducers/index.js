import {combineReducers} from 'redux';
import students from './students'
import errors from './errors'
import auth from './auth'

export default combineReducers({
  students,
  auth,
  errors
});
