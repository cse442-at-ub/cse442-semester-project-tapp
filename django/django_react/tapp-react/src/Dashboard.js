import React, { Component } from 'react';

import Tab from 'react-bootstrap/Tab';
import Nav from 'react-bootstrap/Nav';

import Row from 'react-bootstrap/Row'
import Col from 'react-bootstrap/Col';

import "./styles.css";

import { useDispatch, useSelector, connect } from 'react-redux';
import PropTypes from 'prop-types';

import TopBar from './topbar';
import CalendarTab from './CalendarTab';
import QueueTab from './Queues';
import ProfileModal from './Profile';
import Info from './Info';
import { Link, Redirect } from "react-router-dom";
import {loadCustomUser} from './actions/auth';
import {getEvents} from './actions/students';

export class Dashboard extends Component{
  constructor(props) {
    super(props);
    this.state = {
      Queues: false,
      Staff: false,
      profile: false,
      Calendar: true
    };
  }

  toggleProfile = () => {
    this.setState({
      profile: true
    });
  };

  render() { 
   const auth = this.props.isAuthenticated;
   const {authenticate, user} = this.props.authobj;
   if (!auth)
   {
    return <Redirect exact to="/" />
   }
    return (
    <div id="main2">
    <TopBar prof={() => this.setState({profile: true})} myuser={user}/>
    <Row className="align-middle justify-content-md-center"> 
    <h1 style={{color:"#f5f9e9"}} > Welcome to {user.course} , {user.email} </h1>
    </Row>
    <Row >
    <Tab.Container id="left-tabs-example" defaultActiveKey="Calendar">
        <Col style={{backgroundColor:"#F5F9E9", borderRadius:"5px", height:"125px"}} sm={2}>
          <Nav fill type="pills" className="flex-column">
            <Nav.Item >
              <Nav.Link style={{textColor:"white"}} eventKey="Calendar" onClick = {() => this.setState({Calendar:true, Queues: false, Staff: false})}>Calendar</Nav.Link>
            </Nav.Item>
            <Nav.Item>
              <Nav.Link eventKey="Queues" onClick = {() => this.setState({Queues:true, Calendar: false, Staff: false})} >Queues</Nav.Link>
            </Nav.Item>
            <Nav.Item>
              <Nav.Link eventKey="Info" onClick = {() => this.setState({Queues:false, Calendar: false, Staff: true})} > Staff List </Nav.Link>
            </Nav.Item>
          </Nav>
        </Col>
        <Col sm={9}>
          <Tab.Content>
            <Tab.Pane eventKey="Calendar">
	    <CalendarTab course={user.course} instruct={user.instructor} name={user.name} usr={user} />
            </Tab.Pane>
            <Tab.Pane eventKey="Queues">
	    <QueueTab course={user.course}/>
            </Tab.Pane>
            <Tab.Pane eventKey="Info">
	    <Info course={user.course}/>
            </Tab.Pane>
          </Tab.Content>
        </Col>
    </Tab.Container>
    </Row>
    <ProfileModal user={user} show={this.state.profile} onHide={() => this.setState({profile: false})} />
    </div>
    )
  }
}

Dashboard.propTypes = {
        isAuthenticated: PropTypes.bool,
        events: PropTypes.array.isRequired,
        getEvents: PropTypes.func.isRequired,
        authobj: PropTypes.object.isRequired
}

const mapStateToProps = state => ({
  isAuthenticated: state.auth.isAuthenticated,
  events: state.students.events,
  authobj: state.auth
});

export default connect(mapStateToProps, {getEvents} )(Dashboard);
