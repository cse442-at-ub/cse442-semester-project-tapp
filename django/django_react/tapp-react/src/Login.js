import React, { Component } from 'react';
import Button from 'react-bootstrap/Button';
import Form from 'react-bootstrap/Form';
import Modal from 'react-bootstrap/Modal';
import Alert from 'react-bootstrap/Alert';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { getStudents } from './actions/students';
import { login } from './actions/auth';
import { Link, Redirect } from "react-router-dom";

import 'bootstrap/dist/css/bootstrap.min.css'
import "./styles.css";


export class LoginModal extends Component  {

    constructor(props) {
    super(props)
    this.state = {
            signedUp: false,
            signedValid: false,
            isLoading: false,
            loginAlertF: false,
            loginAlertS: false,
            onHide: props.onHide,
            show: props.show
    };

    this.toggleSignedUp = this.toggleSignedUp.bind(this);
    this.toggleSignedUpBool = this.toggleSignedUpBool.bind(this);
    this.toggleSignedValid = this.toggleSignedValid.bind(this);
    this.toggleIsLoading = this.toggleIsLoading.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.toggleLoginAlertFail = this.toggleLoginAlertFail.bind(this);
    this.toggleLoginAlertSucc = this.toggleLoginAlertSucc.bind(this);
  }

  componentDidUpdate(prevProps) {
    const {error, alert} = this.props;
    if(error !== prevProps.error) {
     if(this.props.error.msg.email) {this.setState({emailValid: false,emailMessage: this.props.error.msg.email.join()})}
     else {this.setState({emailValid: true,emailMessage: ""})}
     if(this.props.error.msg.password) {this.setState({passValid: false, passMessage: this.props.error.msg.pass.join()})}
     else {this.setState({passValid:true,passMessage:""})}
    }
  };
  
  toggleSignedUp = () => {
    this.setState({
      signedUp: !this.state.signedUp
    });
  };

  toggleSignedUpBool = bool => {
    this.setState({
      signedUp: bool
    });
  };

  toggleSignedValid = () => {
    this.setState({
      signedValid: !this.state.signedValid
    });
  };

  toggleIsLoading = () => {
    this.setState({
      isLoading: !this.state.isLoading
    });
  };

  toggleLoginAlertSucc = bool => {
    this.setState({
      loginAlertS: true
    });
  };

  toggleLoginAlertFail = bool => {
    this.setState({
      loginAlertF: true
    });
  };

  handleSubmit = event => {
	  const form = 
		  event.currentTarget;
	  if (form.checkValidity() === false) { event.preventDefault(); event.stopPropagation();}
	  else{
            event.preventDefault()
	    this.toggleSignedUp(true);
	    this.props.login(form.elements.email.value, form.elements.password.value);
	    if(!this.props.isAuth && !this.props.isLoad) {
	      this.toggleLoginAlertFail(true);
	    }
	  }
	  this.toggleSignedValid(true);
  };

  render() {
    if (this.props.isAuth) {
      return <Redirect to="/dashboard" />;
    }
  return (
    <Modal onHide = {this.props.onHide} show = {this.props.show} size="lg" centered>
      <Modal.Header closeButton>
        <Modal.Title> Login </Modal.Title>
      </Modal.Header>
      <Modal.Body>
          <Alert variant="danger" show={this.props.isAuth} dismissible>
            <Alert.Heading> Login Failed </Alert.Heading>
          </Alert>
          <Alert variant="success" show={this.state.loginAlertS} dismissible>
            <Alert.Heading> Login Successfull </Alert.Heading>
          </Alert>
	<Form noValid onSubmit={this.handleSubmit} validated = {this.state.signedValid}>
	  <Form.Group controlId="formBasicEmail">
	    <Form.Label> E-mail </Form.Label>
	    <Form.Control name="email" type="email" placeholder="example@domain.com" required/>
	  </Form.Group>
	  <Form.Group controlId="formBasicPassword">
	    <Form.Label> Password </Form.Label>
	    <Form.Control name="password" type="password" placeholder="Password" required/>
	  </Form.Group>
	  <Button variant="primary" type="submit" disabled={this.props.isLoad}> {this.props.isLoad ? 'Loading...':'Submit'} </Button>
	</Form>
      </Modal.Body>
      <Modal.Footer>
        <Button onClick={this.props.onHide}> Close </Button>
      </Modal.Footer>
    </Modal>
  );
  }
}

LoginModal.propTypes = {
        login: PropTypes.func.isRequired,
        isLoading: PropTypes.bool,
        isAuth: PropTypes.bool,
        error: PropTypes.object.isRequired
}

const mapStateToProps = state => ({
  students: state.students.students,
  isLoad: state.auth.isLoading,
  isAuth: state.auth.isAuthenticated,
  error: state.errors
});

export default connect(mapStateToProps, {login})(LoginModal);
