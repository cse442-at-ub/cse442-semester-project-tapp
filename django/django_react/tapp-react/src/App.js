import React, { Component } from 'react';
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Link
} from "react-router-dom";
import "./styles.css";
import LandingPage from './LandingPage.js';
import Dashboard from './Dashboard.js';
import PRoute from './Proute.js';
import {loadCustomUser} from './actions/auth';

export class App extends Component{
  componentDidMount() {
    this.props.store.dispatch(loadCustomUser());
  }

  render() { 
    return (
	<>
	<Router>
	<Switch>
          <Route exact path="/">
	  <LandingPage />
          </Route>
          <Route exact path="/dashboard" component={Dashboard} >
            <Dashboard />
          </Route>
        </Switch>
	</Router>
	</>
    );
  }
}

export default (App);
