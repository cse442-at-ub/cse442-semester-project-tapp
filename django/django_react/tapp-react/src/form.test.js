import {App} from '../src/App'
import Form from 'react-bootstrap/Form';
import Modal from 'react-bootstrap/Modal';
import {LoginModal} from '../src/Login'
import {SignupModal} from '../src/Form'
import React from 'react';
import renderer from 'react-test-renderer';
import {initialState} from '../src/reducers/students'
import { Provider } from 'react-redux';
import Adapter from "enzyme-adapter-react-16";
import Enzyme, { shallow, mount} from "enzyme";

Enzyme.configure({adapter : new Adapter() });

describe('SignupModal', () => {
  it('renders correctly', () => {
    const mockLoginfn = jest.fn();
    const wrapper = renderer.create(<LoginModal login={mockLoginfn} students={[]} error={{}}/>);
    let tree = wrapper.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('SignupModal renders correctly', () => {
    const mockLoginfn = jest.fn();
    const wrapper = renderer.create(<SignupModal register={mockLoginfn} addEvent={mockLoginfn} error={{}} />);
    let tree = wrapper.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('SignupModal initializes states correctly', () => {
    const mockLoginfn = jest.fn();
    const wrapper = shallow(<SignupModal register={mockLoginfn} addEvent={mockLoginfn} error={{}} />);
    const componentInstance = wrapper.instance();
    expect(wrapper.state('emailValid')).toBe(false);
    expect(wrapper.state('nameValid')).toBe(false);
    expect(wrapper.state('passValid')).toBe(false);
    expect(wrapper.state('first')).toBe(false);
    expect(wrapper.state('isLoading')).toBe(false);
    componentInstance.toggleIsLoading();
    expect(wrapper.state('isLoading')).toBe(true);
  });

  it('Toggling works for similar toggles', () => {
    const mockLoginfn = jest.fn();
    const wrapper = shallow(<SignupModal register={mockLoginfn} addEvent={mockLoginfn} error={{}} />);
    const componentInstance = wrapper.instance();
    expect(wrapper.state('first')).toBe(false);
    expect(wrapper.state('isLoading')).toBe(false);
    componentInstance.toggleIsLoading();
    expect(wrapper.state('isLoading')).toBe(true);
  });

  it('Fails on invalid email', () => {
    const mockLoginfn = jest.fn();
    const wrapper = mount(<SignupModal register={mockLoginfn} show={true} addEvent={mockLoginfn} error={{}} />);
    const componentInstance = wrapper.instance();
    wrapper.find('[name="email"]').first().simulate("change",{target: {value : "a@c"} });
    wrapper.find('[name="name"]').first().simulate("change",{target: {value : "a@c"} });
    wrapper.find('[name="password"]').first().simulate("change",{target: {value : "a@c"} });
    wrapper.find('[name="course"]').first().simulate("change",{target: {value : "a@c"} });
    expect(wrapper.state('emailValid')).toBe(false);
    wrapper.find('[type="submit"]').first().simulate("click",componentInstance.handleSubmit);
    setTimeout(expect(wrapper.state('emailValid')).toBe(false), 2000);
  });

  it('Check validity', () => {
    const mockLoginfn = jest.fn();
    const wrapper = mount(<SignupModal register={mockLoginfn} show={true} addEvent={mockLoginfn} error={{}} />);
    const componentInstance = wrapper.instance();
    wrapper.find('[name="email"]').first().simulate("change",{target: {value : "a@c"} });
    wrapper.find('[name="name"]').first().simulate("change",{target: {value : "a@c"} });
    wrapper.find('[name="password"]').first().simulate("change",{target: {value : "a@c"} });
    wrapper.find('[name="course"]').first().simulate("change",{target: {value : "a@c"} });
    expect(wrapper.state('emailValid')).toBe(false);
    wrapper.find('[type="submit"]').first().simulate("click",componentInstance.handleSubmit);
    expect(componentInstance.checkValidity()).toBe(false);
  });
});
