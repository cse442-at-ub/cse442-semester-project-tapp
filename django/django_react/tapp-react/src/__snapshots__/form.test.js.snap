// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupModal SignupModal renders correctly 1`] = `null`;

exports[`SignupModal renders correctly 1`] = `null`;
